cluster/engine.go

```diff
@@ -53,6 +55,14 @@ type Engine struct {
 	eventHandler    EventHandler
 	healthy         bool
 	overcommitRatio int64
+	ScheduleQueue   *list.List
+}
+
+//ScheduledItem FIX ME : works for affinities as it needs the conatainer name and Image
+//Need to add more types to support all filters
+type ScheduledItem struct {
```

Comment:
I had thought about keeping ```scheduled Item in node.go``` , but doesn't it create cyclic dependency . As ```node.go imports cluster.Engine``` and again we have to remove the item from the queue once the container is scheduled so inside ```engine.create``` we have to do ```node.removefromqueue```. for which engine has to import scheduler/node .  To enforce that scheduled item should go from Node.go add an Image field and container name field in the [swarm/container.go] (https://github.com/docker/swarm/blob/master/cluster/container.go#L8) so that we can avoid the seperate stucts for scheduling . I'll try updating container.go and see if it works and update the PR